jetty-websocket/jetty-websocket-tests/src/test/java/org/eclipse/jetty/websocket/tests/client/BadNetworkTest.java

```diff
@@ -116,14 +144,24 @@ public void testAbruptClientClose() throws Exception
         // Validate that we are connected
         future.get(30,TimeUnit.SECONDS);
 
+        WebSocketSession serverSession = serverSessionRef.get();
+
         // Have client disconnect abruptly
         Session session = wsocket.getSession();
+        LOG.info("client.disconnect");
         session.disconnect();
 
         // Client Socket should see a close event, with status NO_CLOSE
         // This event is automatically supplied by the underlying WebSocketClientConnection
         // in the situation of a bad network connection.
         wsocket.assertReceivedCloseEvent(5000, is(StatusCode.NO_CLOSE), containsString(""));
+
+        TimeUnit.SECONDS.sleep(1); // Let server side close connection
```

Comment:
Tests that need this kind of wait are fragile!
Why can't the await below just wait 10s with the expectation it will see the close as soon as it happens?

Reply:
@gregw because currently there is no way to be notified of the connection close event.
In 9.4.x there are `Connection.Listener`s on the client, but not on the server.

Reply:
i know in 10.0.x, that all the `Connection.Listener` beans added onto the `Connector` will be also added to the `WebSocketConnection` as it is created in the `RFC6455Handshaker` not sure if this feature is present in 9.4.x yet

Reply:
@sbordet but the `serverCloseLatch.await(1, TimeUnit.SECONDS)` in the very next line will wait until the close has been notified.  if the `1` was changed to a `2` then the outcome would be the same??   So I still don't see why a sleep is needed when the next line is waiting on a latch

Reply:
@gregw nope. The `serverCloseLatch` is counted down when the session close event is notified.
We want to verify that that event has been emitted by waiting on the latch.
But then we test if the server-side connection is closed, and for that we have no event so we need to wait 1 second as the test does.

Reply:
I'll backport the `jetty-10.0.x` Connection.Listener support for any websocket server to `jetty-9.4.x` so we can get past this.

Reply:
Done. now this test uses the WebSocket server Connection.Listener (added in commit e56ef9d8f86f436b3ee69c41d970e77682207722)